### DISCUSSION TITLE
EssentialsX 2.19.6 - The (Not Too) Wild Update

### DISCUSSION THREAD

**Discussion by mdcfe** [EssentialsX/Essentials#5053] [Announcements]
**EssentialsX 2.19.6 is here!** 🎉 This release features slight additions, support for Minecraft 1.19.2, and several bug fixes. Keep reading to see all the changes!

Thanks to all our contributors who contributed to this release (all of whom are listed at the bottom of this release), as well as to everyone who has contributed to translations on [Crowdin](https://translate.essentialsx.net)!

Finally, a massive thanks to our community members who have assisted countless people on GitHub and MOSS, including @triagonal, @JasonHorkles, @Evidentsinger14, @Chew, @Laarryy and many others.

In addition, a shoutout to our supporters on [Patreon](https://www.patreon.com/essentialsx), [GitHub Sponsors](https://github.com/sponsors/EssentialsX/) and [Ko-fi](https://ko-fi.com/essentialsx). A special thanks goes to our Hero tier supporters:

- AKP
- @aternosorg
- @pebblehost

Want to be listed here? You can join our Hero tier on Patreon or GitHub at the links above!

Your support helps us to continue spending hours of our free time working on and improving EssentialsX for everyone. If you've got any spare change, you can click one of the buttons below to support the EssentialsX project:

[![Patreon](https://user-images.githubusercontent.com/1917406/125179860-aca4a400-e1ea-11eb-858b-816849f542bf.png)](https://www.patreon.com/essentialsx) [![Ko-fi](https://user-images.githubusercontent.com/1917406/125179861-af06fe00-e1ea-11eb-9e32-73ca5c97939a.png)](https://ko-fi.com/essentialsx) [![GitHub Sponsors](https://user-images.githubusercontent.com/1917406/125179862-b0d0c180-e1ea-11eb-801c-cbd52dfa9465.png)](https://github.com/sponsors/EssentialsX/)

## Supported server versions

EssentialsX officially supports the CraftBukkit, Spigot, and [Paper](https://papermc.io) server software. Specifically, EssentialsX is designed to work on the following Minecraft versions:

* ✅ `1.18.2` and `1.19.2` - EssentialsX actively develops against and supports these versions.
* ⚠️ `1.8.8`, `1.9.4`, `1.10.2`, `1.11.2`, `1.12.2`, `1.13.2`, `1.14.4`, `1.15.2`, `1.16.5`, and `1.17.1` - these versions are still supported, but are not a priority for us, and may be dropped in a future release.

**We recommend using the latest version of Paper**. EssentialsX includes significant improvements that require Paper, including:

* Paper's async chunk loading reduces lag spikes during player teleports.
  CraftBukkit and Spigot do not support async chunk loading from plugins.
* `/anvil`, `/grindstone`, `/cartographytable`, `/loom`, `/smithingtable`, `/stonecutter` are only supported on Paper.
  CraftBukkit and Spigot do not support the API needed for these.
* `/recipe` on 1.12+ requires Paper to avoid a dupe bug introduced by CraftBukkit and Spigot.
* Paper properly shows colours in EssentialsX's console log messages, including EssentialsX Chat messages.
* Paper contains countless overall performance optimisations over CraftBukkit and Spigot.
* Paper supports all properly-written plugins for CraftBukkit and Spigot.

EssentialsX does **not** support the following Minecraft versions or server software:

* 🛑 "Ultra-performance" Paper forks including Akarin, Yatopia, and similar projects - these often perform non-standard behaviour and can break your worlds and installed plugins.
* 🛑 Mohist - **[The Mohist project is known to be sketchy, and has, in the past, tricked users into deleting official EssentialsX jars and installing an unofficial build.](https://essentialsx.net/do-not-use-mohist.html)** We cannot guarantee the safety or functionality of unofficial builds of EssentialsX; you should avoid using Mohist where possible.
* 🛑 Other Forge/Bukkit hybrid servers including Cauldron, Thermos, Magma, CatServer - EssentialsX is built for the Bukkit API, which does not properly support mods. For modded servers, use [SpongeForge](https://www.spongepowered.org/) and [Nucleus](https://nucleuspowered.org/), or an alternative mod for [Forge or Fabric](https://www.curseforge.com/minecraft/mc-mods).
* 🛑 `1.7.10` or below.

## Upgrading to 2.19.6

If you're upgrading from **2.19.3 or above**, you can keep your current EssentialsX config file. There have been no changes to the configuration since that release.

Upgrading from 2.19.2 or older versions of EssentialsX? See the [previous update changelogs](https://github.com/EssentialsX/Essentials/releases).

## Important change: EssentialsX Chat now supports secure chat (#5030)

Minecraft 1.19.1 introduced a new Player Reporting feature on the client, allowing players to report chat messages to Mojang in-game. The server now has a secure chat system which allows players to preview and "sign" their chat messages, which ensures players can't use client mods to fabricate reports with fake messages. If the game can't verify a message, it will show a yellow warning indicator.

As of 2.19.6, EssentialsX Chat supports chat previews. Once you enable chat previews in your `server.properties` file, EssentialsX Chat will send back previews to players so they can send secure "signed" messages without the yellow warning indicator.

Players will need to accept previews when they join the server. They will also need to wait a second between finishing their message and pressing Enter to ensure their message gets previewed and signed.

### Anchor charges deplete on Spigot even when EssentialsX Spawn overrides the spawn location

On CraftBukkit and Spigot, anchor charges will deplete regardless of the `respawn-at-anchor` due to an [unfixed bug in CraftBukkit](https://hub.spigotmc.org/jira/browse/SPIGOT-5989). This bug has been patched in Paper.

### Vanilla signed chat commands may show EssentialsX formatting on 1.19.1+

If you use any vanilla commands that send secure chat messages (including `/minecraft:msg` or `/minecraft:me`), you may see formatting from EssentialsX Chat in these messages. This is due to an implementation quirk in Spigot, which forces plugins to handle all secure previews for vanilla chat commands as if they were public chat messages. There is no known workaround for this at this time, but you can continue to use unsigned commands from EssentialsX and other plugins as normal.

### EssentialsX now supports 1.19.2 (#4932, #4966, #5026, #5027, #5035)

EssentialsX now has full support for Minecraft 1.19.2, including support for the new Wild Update items in `/give`, `/item` and kits, new mobs, and new NMS/OBC mappings for accessing internal CraftBukkit code.

EssentialsX also no longer warns when installed on 1.19.2 servers.

### Other additions

* Translations have been updated with the latest community contributions on Crowdin. (#4823)
* Console colors are once supported again on Paper. (#4941)
* Reduced the amount of user lookups during LuckPerms context calculations. (#4869)
* The `/playtime` command will now no longer appear to progress for vanished users. (#4918)

### Bugs fixed from 2.19.4 and older versions

* The `/tpa` command no longer double charges a user with `/tpauto on`. (#4862)
* The `/nick off` command can now be used by users with only `essentials.nick.changecolors` permissions. (#4876)
* The `TeleportWarmupEvent` event now has the correct teleporter. (#4913)
* Users who are flying while afk while `freeze-afk-players` is set to `true` will no longer be teleported to the ground. (#4944)
* Path and farmland blocks are no longer considered solid blocks. (#4915)
* The `minecraft:` namespace no longer gives an error in the `/give` command. (#4960)
* The `/unlimited` command no longer floods user's inventory when changing stack sizes. (#4877)
* Users who stop being afk no longer fail to count towards the sleep count in some cases. (#4985)
* The `/nick` command now shows your nickname even with `change-displayname` disabled in the config. (#4980)
* `/ess dump` no longer reports offline mode for servers behind Velocity on Paper 1.19 and above. (#5052)
* **Discord:** Console messages in Paper 1.18.2+ properly filter ANSI color codes once again. (#4942)
* **Discord:** Numerous discord formatting issues have been fixed in relation to slash commands. (#4945)
* **Discord:** Roles defined in the `allowed-roles` section of the config will no longer resolve to roles with the same name as another role's id. (#4983) 
* **Discord:** Leave messages are no longer sent to discord when the user was vanished. (#5009)

### Regressions fixed from 2.19.5 and 2.19.6 dev builds

* Essentials no longer fails to load on versions 1.8.8 through 1.12.2. (#4975)
* Local chat allows you to send messages that are 1 character long without sending in public chat. (#5041)

## Changes for developers

There are no API changes in this release!

---

*Note: 2.19.5 had an issue with proxy detection on Paper. 2.19.6 has been released to address this issue.*

For a full commit log, [click here](https://github.com/EssentialsX/Essentials/compare/2.19.4...2.19.6).

<hr /><em>This discussion was created from the release <a href='https://github.com/EssentialsX/Essentials/releases/tag/2.19.6'>EssentialsX 2.19.6 - The (Not Too) Wild Update</a>.</em>

**Comment by ryan-muller** [comment]
Hi, sorry if this is a dumb question, I'm very new to adding plugins to servers and wanted to know if this newest update will work in a 1.19 world? Or is it specifically for 1.19.2?
Excellent plugin by the way. Thank you for your hard work!

**Reply by pop4959** [reply]
Only the latest minor version of each release is officially supported.

**Reply by ryan-muller** [reply]
Thank you so much for your quick reply!
